### PR TITLE
Delete custom iomgr

### PR DESCRIPTION
**Pull request by drfloob**
Forked from @gnossen's https://github.com/grpc/grpc/pull/28276. After gevent's reimplementation, there are no further dependencies on custom iomgr.

### CODE REVIEW

**Review by gnossen — APPROVED** [review]
LGTM in principle. Let's wait until Monday for merge to avoid any issues in case the gevent PR needs to be rolled back.

**Review by markdroth — APPROVED** [review]
Looks great!

**Comment by drfloob** [comment]
CI: `Basic Tests Ruby Linux` is a timeout.

**Pull request merged**
